docs/tutorials/quesion-answering-bot.md

```diff
@@ -0,0 +1,217 @@
+# Question-Answering Chatbot for documentations 
+
+```{article-info}
+:avatar: avatars/gregor.jpg
+:avatar-link: https://jobs.jina.ai
+:avatar-outline: muted
+:author: Gregor @ Jina AI
+:date: November 3, 2021
+```
+
+This tutorial will take you through the process of creating your own question-answering chatbot. 
+This is an inherently difficult task, due to the fuzzyness of human language and the infinite number of questions one could ask. </br> 
+
+One way to solve this is by predicting answers using a neural network that was trained on pairs of quesitons and their corresponding answers. In many cases such a dataset is not available, like in the case of most software documentation. Let's say we want to build a chatbot to answer questions about the jina documentation. What if I told you that there is a way to reframe this task as a search problem and that this would alleviate the need for a large dataset of matching questions and answers? </br>
+
+How you ask? <em>Let me explain!</em>
+
+## Overview 
+Our approach to the problem leverages the [Doc2query method](https://arxiv.org/pdf/1904.08375.pdf), which, form a piece of text, predicts different questions the text could potentially answer. For example, given a sentence such as `Jina is an open source framework for neural search.`, the model predicts questions such as `What is jina?` or `Is jina open source?`. </br>
+The idea here is to predict a number of questions for every part of the original text document, in our case the jina documentation. Then we use an encoder to create a vector representation for each of the predicted questions. These representations are stored and provide the index for our body of text. When a user prompts the bot with a question, we encode it in the same way we encoded our generated questions. Now we can run a similarity search on the encodings. The encoding of the user's query is compared with the encodings in our index to find the closes match. </br>
+Since we know what part of the original text was used to generate the question, that was most similar to the user's query, we can return the original text as an answer to the user. </br> 
```

Comment:
```suggestion
Our approach to the problem leverages the [Doc2query method](https://arxiv.org/pdf/1904.08375.pdf), which, form a piece of text, predicts different questions the text could potentially answer. For example, given a sentence such as `Jina is an open source framework for neural search.`, the model predicts questions such as `What is jina?` or `Is jina open source?`.

The idea here is to predict a number of questions for every part of the original text document, in our case the jina documentation. Then we use an encoder to create a vector representation for each of the predicted questions. These representations are stored and provide the index for our body of text. When a user prompts the bot with a question, we encode it in the same way we encoded our generated questions. Now we can run a similarity search on the encodings. The encoding of the user's query is compared with the encodings in our index to find the closes match.

Since we know what part of the original text was used to generate the question, that was most similar to the user's query, we can return the original text as an answer to the user.
```